OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -64,6 +64,18 @@ protected virtual Activity InnerTick(Actor self, AttackBase attack)
 			if (!attack.Info.IgnoresVisibility && type == TargetType.Actor && !Target.Actor.HasTrait<FrozenUnderFog>() && !self.Owner.CanTargetActor(Target.Actor))
 				return NextActivity;
 
+            // Drop the target if it is being captured by our own team.
+            if(Target.Actor.HasTrait<Building>() && Target.Actor.HasTrait<Capturable>())
+            {
+                Capturable capturable = Target.Actor.TraitOrDefault<Capturable>();
+                CapturesInfo capturesinfo = Target.Actor.TraitOrDefault<CapturesInfo>();
```

Comment:
Do those before and check against null to avoid expensive and unnecessary trait lookups.